src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,714 @@
+        require(sellT1 >= crossPrice, "ARM: sell price too low");
+        require(buyT1 < crossPrice, "ARM: buy price too high");
+
+        traderate0 = PRICE_SCALE * PRICE_SCALE / sellT1; // base (t0) -> token (t1);
+        traderate1 = buyT1; // token (t1) -> base (t0)
+
+        emit TraderateChanged(traderate0, traderate1);
+    }
+
+    /**
+     * @notice set the price that buy and sell prices can not cross.
+     * That is, the buy prices must be below the cross price
+     * and the sell prices must be above the cross price.
+     * If the cross price is being lowered, there can not be any base assets in the ARM. eg stETH.
+     * The base assets should be sent to the withdrawal queue before the cross price can be lowered.
+     * The cross price can be increased with assets in the ARM.
+     * @param newCrossPrice The new cross price scaled to 36 decimals.
+     */
+    function setCrossPrice(uint256 newCrossPrice) external onlyOwner {
+        require(newCrossPrice >= PRICE_SCALE - MAX_CROSS_PRICE_DEVIATION, "ARM: cross price too low");
+        require(newCrossPrice <= PRICE_SCALE, "ARM: cross price too high");
+
+        // If the new cross price is lower than the current cross price
+        if (newCrossPrice < crossPrice) {
+            // Check there is not a significant amount of base assets in the ARM
+            require(IERC20(baseAsset).balanceOf(address(this)) < MIN_TOTAL_SUPPLY, "ARM: too many base assets");
+        }
+
+        crossPrice = newCrossPrice;
+
+        emit CrossPriceUpdated(newCrossPrice);
+    }
+
+    ////////////////////////////////////////////////////
+    ///         Liquidity Provider Functions
+    ////////////////////////////////////////////////////
+
+    /// @notice Preview the amount of shares that would be minted for a given amount of assets
+    /// @param assets The amount of liquidity assets to deposit
+    /// @return shares The amount of shares that would be minted
+    function previewDeposit(uint256 assets) external view returns (uint256 shares) {
+        shares = convertToShares(assets);
+    }
+
+    /// @notice deposit liquidity assets in exchange for liquidity provider (LP) shares.
+    /// The caller needs to have approved the contract to transfer the assets.
+    /// @param assets The amount of liquidity assets to deposit
+    /// @return shares The amount of shares that were minted
+    function deposit(uint256 assets) external returns (uint256 shares) {
+        shares = _deposit(assets, msg.sender);
+    }
+
+    /// @notice deposit liquidity assets in exchange for liquidity provider (LP) shares.
+    /// This function is restricted to the Zap contract.
+    /// @param assets The amount of liquidity assets to deposit
+    /// @param liquidityProvider The address of the liquidity provider
+    /// @return shares The amount of shares that were minted
+    function deposit(uint256 assets, address liquidityProvider) external returns (uint256 shares) {
```

Comment:
⚪️ We should change `liquidityProvider` on both `deposit` and `_deposit()` to either  `to` or `receiver`. This would be consistent other protocols / what we've done in the past.

Reply:
@clement-ux can you make this change

Reply:
Done in this commit https://github.com/OriginProtocol/arm-oeth/commit/dd2b6a865ba9db23c8552f9262bc306ec7f61d9a.